new/detector/implementation/java/.snapshots/TestJavaObjects-object_class

```diff
@@ -0,0 +1,57 @@
+- position: "1:1"
+  content: |-
+    public class User
+    {
+        public String name;
+
+        public String LowercaseName()
+        {
+            return name.toLowerCase();
+        }
+    }
+  data:
+    properties:
+        - name: User
+          node: {}
+          object:
+            detectortype: object
+            matchnode: {}
+            data:
+                properties:
+                    - name: name
+                      node: {}
+                      object:
+                        detectortype: object
+                        matchnode: {}
+                        data: null
+                isvirtual: false
+    isvirtual: false
+- position: "1:1"
+  content: |-
+    public class User
+    {
+        public String name;
+
+        public String LowercaseName()
+        {
+            return name.toLowerCase();
+        }
+    }
+  data:
+    properties:
+        - name: User
+          node: {}
+          object:
+            detectortype: object
+            matchnode: {}
+            data:
+                properties:
+                    - name: LowercaseName
```

Comment:
Not related to this change, but something I can see now you've added the tests:

It would be nice if these properties were both on the same object, rather than having two objects with a single property.

Reply:
fixed, ps. It feels weird to define an object name inside the property as a field called `name` feels like I'm naming property while I'm naming an object.
would be better if we could define object name once and `property` define their own names, or picks them up from their `Object` property.
```
	object := generictypes.Object{
		IsVirtual: false,
	}

	for _, property := range properties {
		object.Properties = append(object.Properties, generictypes.Property{
			Name:   className,
			Node:   node,
			Object: property,
		})
	}
```

would be better if it was something like this
```
	object := generictypes.Object{
		IsVirtual: false,
		Name: className,
	}

	for _, property := range properties {
		object.Properties = append(object.Properties, generictypes.Property{
			Node:   node,
			Object: property,
		})
	}
```

Reply:
Once you have nested objects, there's not a distinction between object and property name. It's both depending on how you look at it. 

We could add a special case for the top level but I don't think it's worth it. You also have anonymous objects where there isn't a name. eg. `someCall({ key: value })`.

Looking at the code again, I think there's still something wrong. There should be a single property for the class at the top level:

```
{ User: { name, LowercaseName } }
```

But it's currently:

```
{ User: { name }, User: { LowercaseName } }
```

Reply:
yeah ok you're right. Could you write the code for me, I honestly don't understand how this property thing works.

As far as I can tell from the code it belongs to the same object with multiple properties, making me think the code is ok but suspect something is wrong with the way we export things. 
Maybe it's just some code gotcha.